contracts/cw1-subkeys/src/contract.rs

```diff
@@ -507,477 +614,388 @@ mod tests {
 
     #[test]
     fn query_allowance_works() {
-        let mut deps = mock_dependencies(&[]);
-
-        let owner = "admin0001";
-        let admins = vec![owner, "admin0002"];
-
-        let spender1 = "spender0001";
-        let spender2 = "spender0002";
-        let spender3 = "spender0003";
-        let initial_spenders = vec![spender1, spender2];
-
-        // Same allowances for all spenders, for simplicity
-        let denom1 = "token1";
-        let amount1 = 1111;
-
-        let allow1 = coin(amount1, denom1);
-        let initial_allowances = vec![allow1.clone()];
-
-        let expires_never = Expiration::Never {};
-        let initial_expirations = vec![expires_never, expires_never];
-
-        let info = mock_info(owner, &[]);
-        setup_test_case(
-            deps.as_mut(),
-            &info,
-            &admins,
-            &initial_spenders,
-            &initial_allowances,
-            &initial_expirations,
-        );
+        let Suite { deps, .. } = SuiteConfig {
```

Comment:
I see the idea of the Suite to handle many things more clearly.
Somehow the usage seems more verbose than the older setup code.

I wonder if it is possible to make usage more compact. With eg. 2 helpers:

`Spender::with_allowance` and `Spender::with_expiring_allowance`. That would reduce those 4-5 line struct inits into 1 line, and then the usage of the new Suite would really shine

Reply:
I am actually thinking about this - this is very generic proposal, but suggestions are very much welcome. I even considered to split initialization and early execution, adding utility for executes, to achieve something like:
```
let Suite { deps, owner_info: owner, .. } = SuiteConfig {
  // Only minimal config, or maybe just `Suite::init(...)` with args required to setup properly
}
.init()
.execute_increase_allowance(&owner, SENDER1, vec![coin(1111, TOKEN)], None)
.execute_increase_allowance(&owner, SENDER2, vec![coin(2222, TOKEN)], None);
```

This would benefit also redability of test logic imho. There can be argument, that this thing hides what is actually called (the `execute` itself), but wrapping it in init hides it even more, so I am ok with that.